catalog/task/update-ocp-tag/0.1/update-ocp-tag.yaml

```diff
@@ -0,0 +1,59 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: update-ocp-tag
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to update pull-spec tag with
+    valid OCP version from get-ocp-version task.
+  params:
+    - name: fromIndex
+      description: The source Index image (catalog of catalogs) FBC fragment
+      type: string
+    - name: targetIndex
+      description: Index image (catalog of catalogs) the FBC fragment will be added to
+      type: string
+    - name: binaryImage
+      description: OCP binary image to be baked into the index image
+      type: string
+    - name: ocpVersion
+      description: OCP version tag to replace the current set tags on index images
+      type: string
+  results:
+    - name: updated-fromIndex
+      description: source Index image (catalog of catalogs) FBC fragment with updated tag
+    - name: updated-targetIndex
+      description: Index image (catalog of catalogs) the FBC fragment will be added to with updated tag
+    - name: updated-binaryImage
+      description: OCP binary image to be baked into the index image with updated tag
+  steps:
+    - name: update-ocp-tag
+      image: quay.io/hacbs-release/release-utils:4d8649dbb2b626f5fe9f4ff83c1bc3be268fad31
+      script: |
+       #!/usr/bin/env sh
+
+        # Function to replace tag in an image
+        replace_tag() {
+            local image="${1}"
+            local oldTag=$(echo "$image" | sed -E 's/.*([vV][0-9]+\.[0-9]{2})$/\1/')
+            local updatedImage="${1%:*}:$(params.ocpVersion)"
+            echo "$updatedImage"
+        }
+
+        # Access the updated image
+        updatedFromIndex=$(replace_tag "$(params.fromIndex)")
+        updatedTargetIndex=$(replace_tag "$(params.targetIndex)")
+        updatedBinaryImage=$(replace_tag "$(params.binaryImage)")
+
+        echo "Updated values"
+        echo -n "$updatedFromIndex" | tee $(results.updated-fromIndex.path)
+        echo    # Print a new line
```

Comment:
I think these comments are redundant. You can remove them.

Reply:
@theflockers they were added because of this thread you created :) https://github.com/redhat-appstudio/release-service-bundles/pull/175#discussion_r1311813663

Reply:
+1 @mmalina

Reply:
The echo is fine. Just saying that it is pretty clear what this `echo` it does and that the comments are unnecessary, but this is opinion. In Germany with my opinions + 2.80 I can ride around by bus.

Reply:
Ah, I read "comments" as "commands" :)